### PR TITLE
[ASCII-732] Bump go to 1.20.8

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Bump Go version to 1.20.8.

### Motivation

Fix some CVEs and stay up-to-date

### Describe how to test/QA your changes

Usual dashboards monitoring

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: tasks/go.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def check_mod_tidy(ctx, test_folder="testmodule"):
         errors_found = []
         for mod in DEFAULT_MODULES.values():
             with ctx.cd(mod.full_path()):
-                ctx.run("go mod tidy -compat=1.17")
```
Comment (amenasria) [inline tasks/go.py:376]: What's the link with bumping Go to `1.20.8` ?

**Review by amenasria — APPROVED** [review]
Left a comment, otherwise LGTM as long as the tests pass !!

File: tasks/go.py (hunk 1 above)
Comment (pgimalac) [reply tasks/go.py:376]: None, but I asked @KSerrania if we wanted to keep that (seemed weird to me to have a 1.17 hardcoded) and the answer was to remove it

Comment (amenasria) [reply tasks/go.py:376]: Ahahah @KSerrania approval is divine, hence LGTM !

**Review by hestonhoffman — APPROVED** [review]

**Review by darccio — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**